### PR TITLE
move MapboxDistanceFormatter to a public package

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

Moves the `MapboxDistanceFormatter` into a public package. We're advertising using this implementation to developers, so we should maintain it in a public package.

### Changelog

```
<changelog>Changed `MapboxDistanceFormatter`'s package from `com.mapbox.navigation.core.internal.formatter` to `com.mapbox.navigation.core.formatter`, making it stable.</changelog>
```

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceFormatter.kt [hunk 1]
```diff
@@ -19,11 +19,9 @@ import java.util.Locale
...
 /**
- * Creates an instance of DistanceFormatter, which can format distances in meters
+ * Implementation of DistanceFormatter, which can format distances in meters
  * based on a language format and unit type.
  *
- * This constructor will infer device language and unit type using the device locale.
- *
  * @param options to build the [MapboxDistanceFormatter]
  */
 class MapboxDistanceFormatter(
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceFormatter.kt:27]: Minor thing 👇 

```kotlin
private companion object {
        private const val smallDistanceUpperThresholdInMeters = 400.0
        private const val mediumDistanceUpperThresholdInMeters = 10000.0
}
```

**Review by Guardiola31337 — COMMENTED** [review]
Is this change to accommodate to `MapboxManeuverApi`? If so, what about also adding `MapboxDistanceFormatter` as default parameter so the API is more user-friendly?

**Comment by LukasPaczos** [comment]
> Is this change to accommodate to MapboxManeuverApi?

Not really, just something I noticed.

We can expose the default param separately :+1:

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceFormatter.kt (hunk 1 above)
Comment (LukasPaczos) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceFormatter.kt:27]: done

**Pull request merged**